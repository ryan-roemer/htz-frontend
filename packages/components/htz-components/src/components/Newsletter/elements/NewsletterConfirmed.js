import React from 'react';
import PropTypes from 'prop-types';
import { FelaComponent, } from 'react-fela';
import { parseComponentProp, } from '@haaretz/htz-css-tools';
import Button from '../../Button/Button'; // eslint-disable-line import/no-named-as-default
import { responsivePropBaseType, } from '../../../propTypes/responsivePropBaseType';
import { newsletterVariantType, } from './types/newsletterVariantType';
import IconClose from '../../Icon/icons/IconClose';

const NewsLetterConfirmedWrapperStyle = ({ theme, variant, }) => ({
  textAlign: 'center',
  extend: [
    parseComponentProp(
      undefined,
      variant,
      theme.mq,
      setWrapperVariant,
      theme.color,
      theme
    ),
  ],
});

function setWrapperVariant(prop, variant, getColor) {
  return {
    backgroundColor: getColor('newsletter', `${variant}Bg`),
    color: getColor('newsletter', `${variant}Text`),
  };
}

const inputUpperNoteStyle = ({ theme, variant, }) => ({
  paddingBottom: '1rem',
  extend: [
    theme.type(-1),
    parseComponentProp(undefined, variant, theme.mq, setVariant, theme.color),
  ],
});

function setVariant(prop, variant, getColor, isError) {
  return {
    color: getColor('newsletter', `${variant}TextTitle`),
  };
}
const closeButtonStyle = {
  display: 'inline-block',
  position: 'absolute',
  left: '0',
  top: '0.5rem',
  paddingTop: '0',
};

const ButtonStyle = {
  marginTop: '1rem',
  type: -1,
  marginBottom: '1rem',
};

NewsletterConfirmed.propTypes = {
  /**
   * Function that handle on close click.
   */
  closeConfirmation: PropTypes.func.isRequired,
  /** determine newsletter host if exists */
  host: PropTypes.oneOf([ 'tm', 'htz', ]),
  /** The Registration status to the newletter */
  signUpStatus: PropTypes.oneOf([ 'ok', 'alredyRegister', 'failed', ]).isRequired,
  /** The TextInput stylistic variant */
  // used for styling
  variant: PropTypes.oneOfType([
    newsletterVariantType,
    PropTypes.arrayOf(
      PropTypes.shape({
        ...responsivePropBaseType,
        value: newsletterVariantType.isRequired,
      })
    ),
  ]),
};

NewsletterConfirmed.defaultProps = {
  host: null,
  variant: 'highlight',
};

export default function NewsletterConfirmed({
  closeConfirmation,
  variant,
  host,
  signUpStatus,
}) {
  return (
    <FelaComponent
      variant={variant}
      rule={NewsLetterConfirmedWrapperStyle}
      render={({ className, theme, }) => {
        const {
          buttons: { newsletterConfirmedButton, },
          texts: { newsletterConfirmedText, newsletterConfirmedTitleText, },
        } = theme.newsletterI18n;
        return (
          <div className={className}>
            <Button
              isFlat
              boxModel={{ hp: 1, vp: 1, }}
              onClick={closeConfirmation}
              variant={theme.newsletterStyle[variant].buttonVariant}
              miscStyles={closeButtonStyle}
            >
              <IconClose size={2} />
            </Button>
            <FelaComponent
              variant={variant}
              rule={inputUpperNoteStyle}
              render="p"
            >
              <strong>{newsletterConfirmedTitleText[signUpStatus]}</strong>
            </FelaComponent>
            <FelaComponent
              variant={variant}
              rule={inputUpperNoteStyle}
              render="p"
            >
              {newsletterConfirmedText[signUpStatus]}
            </FelaComponent>
            <Button
              boxModel={{ hp: 3, vp: 1, }}
              miscStyles={ButtonStyle}
              variant={theme.newsletterStyle[variant].buttonVariant}
              {...(signUpStatus === 'failed'
                ? { onClick: () => closeConfirmation(), }
                : {
                  href:
                      host === 'tm'
                        ? 'https://www.themarker.com/personal-area/newsletter'
                        : 'https://www.haaretz.co.il/personal-area/newsletter',
                })}
            >
              {newsletterConfirmedButton[signUpStatus]}
            </Button>
          </div>
        );
      }}
    />
  );
}
