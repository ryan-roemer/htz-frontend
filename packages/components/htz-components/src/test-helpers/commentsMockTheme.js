const mockTheme = {
  // Constants
  bps: Object.freeze({
    widths: Object.freeze({
      s: 600,
      m: 768,
      l: 1024,
      xl: 1280,
    }),
    misc: Object.freeze({
      landscape: '(orientation: landscape)',
      portrait: '(orientation: portrait)',
      hidpi: '(min-resolution: 1.5dppx)',
    }),
  }),
  btnStyle: Object.freeze({
    // Border width
    borderBottomWidth: 1,
    borderEndWidth: 1,
    borderStartWidth: 1,
    borderTopWidth: 1,

    // Border style
    borderBottomStyle: 'solid',
    borderEndStyle: 'solid',
    borderStartStyle: 'solid',
    borderTopStyle: 'solid',
    radius: 0,
    boxModel: { hp: 4, vp: 1, },

    // Font Style
    fontWeight: 'bold',
  }),
  commentsStyle: Object.freeze({
    textInputVariant: 'primary',
  }),
  inputStyle: Object.freeze({
    borderWidth: 1,
    lines: 1,
    borderStyle: 'solid',
    radius: 0,
    height: 6,
    // typographic Scale from theme
    typeScale: -1,
    fontWeightLabel: 'bold',
  }),
  selectStyle: Object.freeze({
    borderWidth: 1,
    lines: 0,
    borderStyle: 'solid',
  }),
  typeConf: Object.freeze({
    default: Object.freeze({
      base: 16,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 6,
      steps: 5,
    }),
    xl: Object.freeze({
      base: 18,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 7,
      steps: 5,
    }),
  }),

  commentI18n: Object.freeze({
    tags: Object.freeze({
      editorsPick: 'בחירת העורכים',
      usersPick: 'בחירת הגולשים',
    }),
    buttons: Object.freeze({
      readMoreBtnTxt: 'קרא עוד',
      replyBtnTxt: 'הגב',
      reportAbuseBtnTxt: 'דווח כפוגעני',
    }),
  }),

  commentFormI18n: Object.freeze({
    buttons: Object.freeze({
      sendBtnTxt: 'שלח',
      cancelBtnTxt: 'בטל',
      toggleUserBtnText: 'להוספת תגובה מזוהה לחץ כאן',
    }),
    labels: Object.freeze({
      nameLabelTxt: 'שם',
      commentLabelTxt: 'תגובה',
    }),
    notes: Object.freeze({
      nameNoteTxt: 'הזינו שם שיוצג כמחבר התגובה',
      commentNoteTxt: 'בשליחת תגובה זו הנני מצהיר שהינני מסכים/ה עם תנאי השימוש של אתר הארץ',
    }),
    errorNotes: Object.freeze({
      nameErrorNoteTxt: 'חובה להזין שם',
      commentErrorNoteTxt: 'נא להזין את תוכן התגובה',
    }),
  }),
  commentSentI18n: Object.freeze({
    buttons: Object.freeze({
      getNotificationsBtnTxt: 'עדכנו אותי',
      dontGetNotificationsBtnTxt: 'לא תודה',
      closeBtnText: 'סגור',
    }),
    labels: Object.freeze({
      emailLabelTxt: 'דוא"ל',
    }),
    notes: Object.freeze({
      emailNoteTxt: 'הזינו כתובת דוא"ל לקבלת התראות',
    }),
    errorNotes: Object.freeze({
      emailErrorNoteTxt: 'נא להזין כתובת דוא"ל תקינה',
    }),
    texts: Object.freeze({
      commentRecievedBoldText: 'תגובתך נקלטה בהצלחה, ',
      commentRecievedText: 'ותפורסם על פי מדיניות המערכת.',
      commentRecievedTextSecondRow: 'באפשרותך לקבל התראה בדוא"ל כאשר תגובתך תאושר ותפורסם',
      commentRecievedBoldTextThankYouPage: 'תודה!',
      commentRecievedTextThankYouPage: 'תגובתך נקלטה בהצלחה ותפורסם על פי מדיניות המערכת',
    }),
  }),
  commentsSectionI18n: Object.freeze({
    buttons: Object.freeze({
      loadAllCommentsBtnText: 'טען את כל התגובות',
    }),
    selectItems: Object.freeze({
      dateDescendingItemTxt: 'מהאחרונה לראשונה',
      dateAscendingItemTxt: 'מהראשונה לאחרונה',
      commentRatingItemTxt: 'הצג לפי דירוג',
      editorsPickItemTxt: 'בחירת העורכים',
    }),
    texts: Object.freeze({
      chooseSortMethodText: 'סדרו את התגובות',
    }),
  }),

  // Methods
  color: () => 'red',
  getTransition: () => ({
    transitionDuration: undefined,
    transitionTimingFunction: undefined,
    transitionDelay: undefined,
  }),
  mq: () => () => ({
    '@media (min-width: 37.5em) and (max-width: 64em)': {
      color: 'red',
    },
  }),
  type: () => () => ({}),
};

export default mockTheme;
