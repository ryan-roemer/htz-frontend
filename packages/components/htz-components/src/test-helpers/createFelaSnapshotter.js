import React from 'react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/no-extraneous-dependencies
import { renderToString, } from 'fela-tools';

// function Wrapper({children,}) {
//   return children;
// }
export default function createFelaSnapshotter(StyleProvider, testRenderer) {
  return function testFelaSnapshot(component) {
    const testObj = {
      // Output an object with both the component snapshot and raw styles.
      component: renderer
        .create(
          <StyleProvider renderer={testRenderer}>
            <div>{component}</div>
          </StyleProvider>
        )
        .toJSON(),
      styles: renderToString(testRenderer),
    };

    return testObj;
  };
}
