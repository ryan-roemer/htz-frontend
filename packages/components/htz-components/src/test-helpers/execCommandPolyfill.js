/* global document */
/**
 * curruntly jest doesnt support document.execCommand
 * the following function act as a polyfill's for this function, it returns the prev
 * function so we can cancel the polyfill at the end of the test
 */
export default function execCommandPolyfill() {
  const prevExecCommand = document.execCommand;
  document.execCommand = document.execCommand || function noop() {};
  return prevExecCommand;
}
