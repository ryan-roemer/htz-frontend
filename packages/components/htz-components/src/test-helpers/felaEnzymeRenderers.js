// eslint-disable-next-line import/no-extraneous-dependencies
import htzTheme from '@haaretz/htz-theme';
import { createRenderer, createFelaEnzymeRenderers, } from '@haaretz/fela-utils';
// import createFelaEnzymeRenderers from '../utils/createFelaEnzymeRenderers';

const testRenderer = createRenderer({ isRtl: true, });
const { felaMount, felaShallow, } = createFelaEnzymeRenderers(
  testRenderer,
  htzTheme
);

export { felaMount, felaShallow, };
