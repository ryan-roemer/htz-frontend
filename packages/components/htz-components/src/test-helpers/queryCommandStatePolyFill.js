/* global document */

/**
 * curruntly jest doesnt support document.queryCommandState()
 * the following function act as a polyfill's for this function, it returns the prev
 * function so we can cancel the polyfill at the end of the test
 */
export default function queryCommandStatePolyFill() {
  const prevQueryCommandState = document.queryCommandState;
  document.queryCommandState = document.queryCommandState || (() => true);

  return prevQueryCommandState;
}
