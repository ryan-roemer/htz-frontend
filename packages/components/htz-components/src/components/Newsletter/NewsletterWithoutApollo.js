import React from 'react';
import PropTypes from 'prop-types';
import { FelaComponent, } from 'react-fela';
import { parseComponentProp, parseStyleProps, } from '@haaretz/htz-css-tools';
import { stylesPropType, } from '../../propTypes/stylesPropType';
import { responsivePropBaseType, } from '../../propTypes/responsivePropBaseType';
import NewsletterConfirmed from './elements/NewsletterConfirmed';
import { NewsletterForm, } from './elements/newsletterForm';
import { newsletterVariantType, } from './elements/types/newsletterVariantType';
import Dialog from '../A11yDialog/A11yDialog';

const newsletterWrapperStyle = ({ theme, variant, miscStyles, }) => ({
  position: 'relative',
  padding: '2rem',
  width: '100%',
  extend: [
    parseComponentProp(
      undefined,
      variant,
      theme.mq,
      setVariant,
      theme.color,
      theme
    ),
    ...(miscStyles ? parseStyleProps(miscStyles, theme.mq, theme.type) : []),
  ],
});

function setVariant(prop, variant, getColor) {
  return {
    backgroundColor: getColor('newsletter', `${variant}Bg`),
    color: getColor('newsletter', `${variant}Text`),
  };
}

const newsletterConfirmStyle = {
  width: [ { from: 's', value: '99%', }, { until: 's', value: '98%', }, ],
  paddingTop: [ { from: 's', value: '2rem', }, { until: 's', value: '5rem', }, ],
  paddingBottom: [ { from: 's', value: '2rem', }, { until: 's', value: '3rem', }, ],
};

export default class NewsletterWithoutApollo extends React.Component {
  static propTypes = {
    /** The element's DOM ID */
    id: PropTypes.string,
    /** indicates loading state of the data */
    loading: PropTypes.bool.isRequired,
    /** Indicates article category id */
    segmentId: PropTypes.number.isRequired,
    /**
     * registration function.
     */
    signUpNewsletter: PropTypes.func.isRequired,
    /** Indicates registered/paying user email */
    userEmail: PropTypes.string,
    /**
     * A special property holding miscellaneous CSS values that
     * trump all default values. Processed by
     * [`parseStyleProps`](https://Haaretz.github.io/htz-frontend/htz-css-tools#parsestyleprops)
     */
    miscStyles: stylesPropType,
    /** The `<newsletter />`'s stylistic variant */
    variant: PropTypes.oneOfType([
      newsletterVariantType,
      PropTypes.arrayOf(
        PropTypes.shape({
          ...responsivePropBaseType,
          value: newsletterVariantType.isRequired,
        })
      ),
    ]).isRequired,
  };
  static defaultProps = {
    id: PropTypes.string,
    miscStyles: null,
    userEmail: null,
  };
  state = {
    confirmed: false,
    id: null,
  };

  componentWillMount() {
    if (!this.state.id) {
      this.setState({
        id:
          this.props.id ||
          `newsletterRegForm${Math.round(Math.random() * 1000000)}`,
      });
    }
  }

  confirmState = ({ data, }) => {
    if (data.signUpNewsletter) {
      this.setState({
        confirmed: true,
      });
    }
  };

  render() {
    const {
      loading,
      userEmail,
      segmentId,
      signUpNewsletter,
      variant,
      miscStyles,
    } = this.props;
    return (
      <div>
        <FelaComponent
          variant={variant}
          miscStyles={miscStyles}
          rule={newsletterWrapperStyle}
          render={({ className, theme, }) => (
            <div className={className} id={this.state.id}>
              <NewsletterForm
                loading={loading}
                variant={variant}
                userEmail={userEmail}
                segmentId={segmentId}
                signUpNewsletter={signUpNewsletter}
                setParentState={this.confirmState}
              />
              <Dialog
                appendTo={`${this.state.id}DialogWrapper`}
                elementToHide={this.state.id}
                isVisible={this.state.confirmed}
                overlayBgColor={theme.color('newsletter', `${variant}Bg`)}
                containerMiscStyles={newsletterConfirmStyle}
                render={({ handleClose, }) => (
                  <NewsletterConfirmed
                    variant={variant}
                    closeConfirmation={handleClose}
                  />
                )}
              />
              <div id={`${this.state.id}DialogWrapper`} />
            </div>
          )}
        />
      </div>
    );
  }
}
