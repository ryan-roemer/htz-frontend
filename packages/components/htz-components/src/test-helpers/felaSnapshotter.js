import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
