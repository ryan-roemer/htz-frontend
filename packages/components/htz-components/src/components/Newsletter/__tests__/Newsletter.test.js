import React from 'react';
import toJson from 'enzyme-to-json';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import { felaMount, } from '../../../test-helpers/felaEnzymeRenderers';
import NewsletterWithoutApollo from '../NewsletterWithoutApollo';
import NewsletterConfirmed from '../elements/NewsletterConfirmed';

Math.random = jest.fn(() => 0.123456789);
const mockFunc = jest.fn();

describe('<Newsletter />', () => {
  describe('Newsletter DOM element', () => {
    // TODO unskip on Enzyme React16 support
    it.skip('should correctly render a Newsletter initial view', () => {
      const wrapper = felaMount(
        <NewsletterWithoutApollo
          buttonText="הרשמה"
          contentId="7.1234567"
          headlineText="headline text"
          loading={false}
          signUpNewsletter={mockFunc}
          miscStyles={{
            maxWidth: '80rem',
            marginLeft: 'auto',
            marginRight: 'auto',
          }}
          segmentId={1420800}
          variant="highlight"
        />
      );
      expect(toJson(wrapper, { mode: 'shallow', })).toMatchSnapshot();
    });
  });
  describe('NewsletterConfirmed DOM element', () => {
    it('should correctly render a NewsletterConfirmed without any props', () => {
      const { component, styles, } = felaSnapshotter(
        <NewsletterConfirmed closeConfirmation={mockFunc} />
      );
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
  });
});
