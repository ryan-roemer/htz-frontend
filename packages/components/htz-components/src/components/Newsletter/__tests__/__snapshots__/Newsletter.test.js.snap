// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Newsletter /> Newsletter DOM element should correctly render a Newsletter initial view 1`] = `
<NewsletterWithoutApollo
  buttonText="הרשמה"
  headlineText="headline text"
  host={null}
  icon={null}
  id={null}
  loading={false}
  miscStyles={
    Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "80rem",
    }
  }
  onSubmit={[Function]}
  segmentId={1420800}
  signUpNewsletter={[MockFunction]}
  userEmail={null}
  variant="highlight"
/>
`;

exports[`<Newsletter /> NewsletterConfirmed DOM element should correctly render a NewsletterConfirmed without any props 1`] = `
<div>
  <div
    className="htz-a htz-b htz-c"
  >
    <button
      className="htz-d htz-e htz-f htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n htz-o htz-p htz-q htz-r htz-s htz-t htz-u htz-v htz-w htz-x htz-y htz-z htz-ab htz-ac htz-ae htz-af htz-ag htz-ah htz-ai htz-aj htz-ak htz-al htz-am htz-an htz-ao htz-ap htz-aq htz-ar htz-as htz-at htz-au htz-av htz-aw htz-ax htz-ay htz-az htz-ba htz-bb htz-bc htz-bd htz-be htz-bf htz-bg htz-bh htz-bi htz-bj htz-bk htz-bl htz-bm"
      id={null}
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="htz-bn htz-bo htz-bp htz-bq htz-br htz-bs htz-bt"
      >
        <svg
          className="htz-bu htz-bv htz-bw htz-bx"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
        >
           
          <path
            d="M148.8 129l99.6 99.6-19.8 19.8-99.6-99.6-99.6 99.6-19.8-19.8 99.6-99.6L9.6 29.4 29.4 9.6l99.6 99.6 99.6-99.6 19.8 19.8-99.6 99.6z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        className="htz-by htz-bz htz-ca htz-m htz-am htz-cb htz-cc htz-cd htz-ce htz-cf htz-cg htz-ch htz-ci"
      />
    </button>
    <p
      className="htz-cj htz-ck htz-cl htz-cm htz-cn htz-c"
    >
      <strong>
        תודה שנרשמת
      </strong>
    </p>
    <p
      className="htz-cj htz-ck htz-cl htz-cm htz-cn htz-c"
    >
      יש לנו דיוורים נוספים שעשויים לעניין אותך
    </p>
    <a
      className="htz-d htz-e htz-f htz-g htz-h htz-co htz-j htz-k htz-l htz-cp htz-n htz-o htz-p htz-q htz-r htz-s htz-t htz-u htz-v htz-w htz-x htz-cq htz-z htz-ab htz-cr htz-ae htz-af htz-cs htz-ah htz-ai htz-ct htz-ak htz-cu htz-ck htz-cl htz-cm htz-cn htz-cv htz-an htz-ao htz-ap htz-aq htz-cw htz-as htz-at htz-au htz-av htz-aw htz-ax htz-ay htz-cx htz-ba htz-bb htz-bc htz-bd htz-be htz-bf htz-bg htz-cy htz-bi htz-bj htz-bk htz-bl htz-bm"
      href="https://www.haaretz.co.il/personal-area/newsletter"
      id={null}
      onClick={null}
      target={null}
    >
      <span
        className="htz-bn htz-bo htz-bp htz-bq htz-br htz-bs htz-bt"
      >
        לרשימה המלאה
      </span>
      <span
        className="htz-by htz-bz htz-ca htz-m htz-am htz-cb htz-cc htz-cd htz-ce htz-cf htz-cg htz-ch htz-ci"
      />
    </a>
  </div>
</div>
`;

exports[`<Newsletter /> NewsletterConfirmed DOM element should correctly render a NewsletterConfirmed without any props 2`] = `".htz-a{text-align:center}.htz-b{background-color:#FFE70C}.htz-c{color:#222}.htz-d{align-items:center}.htz-e{-webkit-appearance:none}.htz-f{-moz-appearance:none}.htz-g{appearance:none}.htz-h{cursor:pointer}.htz-i{display:inline-block}.htz-j{font-weight:bold}.htz-k{font-size:inherit}.htz-l{justify-content:center}.htz-m{position:absolute}.htz-n{transform:scale(1)}.htz-o{transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-p{text-decoration:none}.htz-q{white-space:nowrap}.htz-r::moz-focus-inner{border:0}.htz-s::moz-focus-inner{padding:0}.htz-t{border-bottom-width:1px}.htz-u{border-bottom-style:solid}.htz-v{padding-bottom:calc(1rem - 1px)}.htz-w{border-top-width:1px}.htz-x{border-top-style:solid}.htz-y{padding-top:0}.htz-z{border-right-width:1px}.htz-ab{border-right-style:solid}.htz-ac{padding-right:1rem}.htz-ae{border-left-width:1px}.htz-af{border-left-style:solid}.htz-ag{padding-left:1rem}.htz-ah{border-radius:0}.htz-ai{background-color:transparent}.htz-aj{border-color:transparent}.htz-ak{color:#2D2D2D}.htz-al{left:0}.htz-am{top:0}.htz-an:visited{color:#2D2D2D}.htz-ao:hover{text-decoration:none}.htz-ap:hover{transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-aq:hover{background-color:rgba(0,0,0,.1)}.htz-ar:hover{border-color:transparent}.htz-as:hover{color:#2D2D2D}.htz-at:hover{-webkit-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-au:hover{-moz-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-av:focus{outline:none}.htz-aw:focus{text-decoration:none}.htz-ax:focus{transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-ay:focus{background-color:#2D2D2D}.htz-az:focus{border-color:transparent}.htz-ba:focus{color:hsl(0,0%,100%)}.htz-bb:focus{-webkit-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-bc:focus{-moz-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-bd:active{outline:none}.htz-be:active{text-decoration:none}.htz-bf:active{transition-duration:0s}.htz-bg:active{background-color:transparent !important}.htz-bh:active{border-color:transparent}.htz-bi:active{color:#2D2D2D}.htz-bj{-webkit-box-align:center}.htz-bk{-webkit-box-pack:center}.htz-bl{-webkit-transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-bm{-moz-transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-bn{opacity:1}.htz-bo{transition-property:opacity}.htz-bp{transition-duration:0.16666666666666666s}.htz-bq{transition-timing-function:cubic-bezier(.55, 0, .1, 1)}.htz-br{transition-delay:.3s}.htz-bs{-webkit-transition-property:opacity}.htz-bt{-moz-transition-property:opacity}.htz-bu{vertical-align:-0.15em}.htz-bv{color:inherit}.htz-bw{fill:#FFF}.htz-bx{font-size:2rem}.htz-by{background-color:rgba(200,200,200, 0.3)}.htz-bz{height:100%}.htz-ca{right:0}.htz-cb{transform:scaleX(0)}.htz-cc{transform-origin:right}.htz-cd{width:100%}.htz-ce{animation-direction:alternate}.htz-cf{animation-iteration-count:infinite}.htz-cg{animation-duration:1s}.htz-ch{animation-delay:0.75s}.htz-ci{animation-timing-function:cubic-bezier(.55, 0, .1, 1)}.htz-cj{padding-bottom:1rem}.htz-co{display:-webkit-inline-box;display:-moz-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-flex;display:inline-flex}.htz-cp{position:relative}.htz-cq{padding-top:calc(1rem - 1px)}.htz-cr{padding-right:3rem}.htz-cs{padding-left:3rem}.htz-ct{border-color:#2D2D2D}.htz-cu{margin-top:1rem}.htz-cv{margin-bottom:1rem}.htz-cw:hover{border-color:#2D2D2D}.htz-cx:focus{border-color:#2D2D2D}.htz-cy:active{border-color:#2D2D2D}@media (max-width: 79.9375em){.htz-ck{font-size:2.3333333333333335rem}.htz-cl{line-height:1.2857142857142858em}}@media (min-width: 80em){.htz-cm{font-size:2.2857142857142856rem}.htz-cn{line-height:1.3125em}}"`;
